packages/k2-alpine/src/components/Avatar/HexagonImageView.tsx

```diff
@@ -0,0 +1,156 @@
+import MaskedView from '@react-native-masked-view/masked-view'
+import React, { useEffect } from 'react'
+import { ImageSourcePropType, Image, Platform } from 'react-native'
+import Animated, {
+  Easing,
+  useAnimatedProps,
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming
+} from 'react-native-reanimated'
+import Svg, { Path } from 'react-native-svg'
+import { alpha } from '../../utils'
+import {
+  colors,
+  darkModeColors,
+  lightModeColors
+} from '../../theme/tokens/colors'
+import { useTheme } from '../..'
+
+export const HexagonImageView = ({
+  source,
+  height,
+  isSelected
+}: {
+  source: ImageSourcePropType
+  height: number
+  isSelected?: boolean
+}): JSX.Element => {
+  const { theme } = useTheme()
+  const selectedAnimation = useSharedValue(0)
+  const selectedAnimatedStyle = useAnimatedStyle(() => ({
+    opacity: selectedAnimation.value
+  }))
+
+  useEffect(() => {
+    selectedAnimation.value = withTiming(isSelected ? 1 : 0, {
+      duration: 200,
+      easing: Easing.inOut(Easing.ease)
+    })
+  }, [isSelected, selectedAnimation])
+
+  return (
+    <MaskedView
+      maskElement={
+        <Svg width={height} height={height} viewBox={hexagonPath.viewBox}>
+          <Path d={hexagonPath.path} fill="black" />
+        </Svg>
+      }>
+      <Image
+        key={`image-${source}`}
+        resizeMode="cover"
+        source={source}
+        style={{ width: height, height: height }}
+      />
+      <Animated.View
+        style={[
+          {
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            right: 0,
+            bottom: 0,
+            backgroundColor: alpha(
+              theme.isDark ? colors.$neutralWhite : colors.$neutral850,
+              0.8
+            ),
+            alignItems: 'center',
+            justifyContent: 'center'
+          },
+          selectedAnimatedStyle
+        ]}>
+        <Arrow key={theme.isDark ? 'dark' : 'light'} isSelected={isSelected} />
+      </Animated.View>
+    </MaskedView>
+  )
+}
+
+export const HexagonBorder = ({ height }: { height: number }): JSX.Element => {
+  const { theme } = useTheme()
+
+  return (
+    <Svg
+      style={{ position: 'absolute', top: 0, left: 0, right: 0, bottom: 0 }}
+      width={height}
+      height={height}
+      viewBox={hexagonPath.viewBox}>
+      <Path
+        d={hexagonPath.path}
+        fill="none"
+        stroke={theme.isDark ? colors.$neutralWhite : 'black'}
+        strokeOpacity={0.1}
+        strokeWidth="1"
+      />
+    </Svg>
+  )
+}
+
+const Arrow = ({ isSelected }: { isSelected?: boolean }): JSX.Element => {
+  const { theme } = useTheme()
+  const arrowAnimation = useSharedValue(
+    Platform.OS === 'ios' ? arrowPath.length : 0
+  )
+  const arrowAnimatedProps = useAnimatedProps(() => ({
+    strokeDashoffset: arrowAnimation.value
+  }))
+
+  useEffect(() => {
+    if (Platform.OS === 'ios') {
```

Comment:
just curious why you only did this for iOS

Reply:
I couldn’t get the arrow animation to work on Android, so I gave up. Let me add a short comment.